compose/ui/ui-test-junit4/src/desktopTest/kotlin/androidx/compose/ui/test/ComposeUiTestTest.kt

```diff
@@ -158,6 +166,31 @@ class ComposeUiTestTest {
         }
     }
 
+    @Test
+    fun testTimeout() {
+        val timeout = 100.milliseconds
+        val error = assertFailsWith<AssertionError> {
+            runComposeUiTest(testTimeout = timeout) {
+                // switch a dispatcher to not skip the delay
+                withContext(Dispatchers.Default) { delay(1000) }
+            }
+        }
+        assertTrue(
+            error.message!!.startsWith("After waiting for $timeout, the test coroutine is not completing")
+        )
+    }
+
+    @Test
+    fun testFailedAssertion() {
+        val error = assertFailsWith<AssertionError> {
+            runComposeUiTest {
+                @Suppress("KotlinConstantConditions")
+                assertTrue(1 == -1)
+            }
+        }
+        assertEquals("Expected value to be true.", error.message)
```

Comment:
Likewise.

Reply:
fixed. Now the tests uses `assertThat(...)` which gives more specific types and properties to verify.